filesystem.go

```diff
@@ -675,10 +728,31 @@ func (fs *filesystem) fetchPodNetwork(podID string) (NetworkNamespace, error) {
 	return NetworkNamespace{}, fmt.Errorf("Unknown network type")
 }
 
+func (fs *filesystem) fetchHypervisorState(podID string, state interface{}) error {
```

Comment:
I'll cast data to []byte, if not, I get this error

```sh
# github.com/containers/virtcontainers
./filesystem.go:735:25: cannot use data (type interface {}) as type []byte in argument to json.Unmarshal: need type assertion
Makefile:27: recipe for target 'build' failed
make: *** [build] Error 2
```

```go
func (fs *filesystem) fetchHypervisorState(podID string, state interface{}) error {
        data, err := fs.fetchResource(true, podID, "", hypervisorFileType)
	if err != nil {
		return err
	}

        if err = json.Unmarshal(data.([]byte), state); err != nil {
                return err
        }

        return nil
}
```

Reply:
@devimc you're right but there is an underlying problem here and it's not related to your PR, but to the way the storage (and the fetch specifically) is implemented in virtcontainers.
I have submitted https://github.com/containers/virtcontainers/pull/472 in order to make this more flexible and it should solve your issues. I am waiting for @sameo to review this PR and hopefully this will get merged soon.

Reply:
#472 and #477 have been merged! You can rebase your PR on top of it, and simplify the storage part :)